src/benchmarklib/tpcds/tpcds_table_generator.cpp

```diff
@@ -0,0 +1,1141 @@
+const auto web_sales_column_names = boost::hana::make_tuple("ws_sold_date_sk", "ws_sold_time_sk", "ws_ship_date_sk", "ws_item_sk", "ws_bill_customer_sk", "ws_bill_cdemo_sk", "ws_bill_hdemo_sk", "ws_bill_addr_sk", "ws_ship_customer_sk", "ws_ship_cdemo_sk", "ws_ship_hdemo_sk", "ws_ship_addr_sk", "ws_web_page_sk", "ws_web_site_sk", "ws_ship_mode_sk", "ws_warehouse_sk", "ws_promo_sk", "ws_order_number", "ws_quantity", "ws_wholesale_cost", "ws_list_price", "ws_sales_price", "ws_ext_discount_amt", "ws_ext_sales_price", "ws_ext_wholesale_cost", "ws_ext_list_price", "ws_ext_tax", "ws_coupon_amt", "ws_ext_ship_cost", "ws_net_paid", "ws_net_paid_inc_tax", "ws_net_paid_inc_ship", "ws_net_paid_inc_ship_tax", "ws_net_profit"); // NOLINT
+
+const auto web_site_column_types = boost::hana::tuple<tpcds_key_t, pmr_string, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<tpcds_key_t>, std::optional<tpcds_key_t>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<int32_t>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<int32_t>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<pmr_string>, std::optional<float>, std::optional<float>>(); // NOLINT
+const auto web_site_column_names = boost::hana::make_tuple("web_site_sk", "web_site_id", "web_rec_start_date", "web_rec_end_date", "web_name", "web_open_date_sk", "web_close_date_sk", "web_class", "web_manager", "web_mkt_id", "web_mkt_class", "web_mkt_desc", "web_market_manager", "web_company_id", "web_company_name", "web_street_number", "web_street_name", "web_street_type", "web_suite_number", "web_city", "web_county", "web_state", "web_zip", "web_country", "web_gmt_offset", "web_tax_percentage"); // NOLINT
+// clang-format on
+}  // namespace
+
+namespace opossum {
+
+TpcdsTableGenerator::TpcdsTableGenerator(uint32_t scale_factor, ChunkOffset chunk_size, int rng_seed)
+    : AbstractTableGenerator(create_benchmark_config_with_chunk_size(chunk_size)) {
+  init_tpcds_tools(scale_factor, rng_seed);
+}
+
+TpcdsTableGenerator::TpcdsTableGenerator(uint32_t scale_factor,
+                                         const std::shared_ptr<BenchmarkConfig>& benchmark_config, int rng_seed)
+    : AbstractTableGenerator(benchmark_config) {
+  init_tpcds_tools(scale_factor, rng_seed);
+}
+
+std::unordered_map<std::string, BenchmarkTableInfo> TpcdsTableGenerator::generate() {
+  const auto max_rows = std::numeric_limits<ds_key_t>::max();
+
+  auto table_info_by_name = std::unordered_map<std::string, BenchmarkTableInfo>();
+
+  table_info_by_name["call_center"].table = generate_call_center(max_rows);
```

Comment:
With `std::numeric_limits<ds_key_t>::max()` being visible all over the file, it seems weird to pass it as an argument here.

Reply:
That's because of the test, where we pass a small number.

Reply:
Got it. As `generate_call_center` already has `max(ds_key_t)` as a default parameter, I would refrain from repeating it here. In the header, please add a comment before `max_rows` is used first explaining that it is used by the tests to limit the generation to a certain number of rows.

Reply:
Done.